bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,595 @@
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.communication-error");
+                retryConnection = connectRetriesRemaining > 0;
+                break;
+
+            case BRIDGE_UNINITIALIZED:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                        "@text/offline.clip2.conf-error-assets-not-loaded");
+                break;
+
+            case NONE:
+            default:
+                updateSelf(); // go online
+                break;
+        }
+
+        // this method schedules itself to be called again in a loop..
+        ScheduledFuture<?> task = checkConnectionTask;
+        if (Objects.nonNull(task)) {
+            task.cancel(false);
+        }
+        int milliSeconds;
+        if (retryApplicationKey) {
+            // short delay used during attempts to create or validate an application key
+            milliSeconds = FAST_SCHEDULE_MILLI_SECONDS;
+            applKeyRetriesRemaining--;
+        } else {
+            // default delay, set via configuration parameter, used as heart-beat 'just-in-case'
+            Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+            milliSeconds = config.checkSeconds * 1000;
+            if (retryConnection) {
+                // exponential back off delay used during attempts to reconnect
+                int backOffDelay = 60000 * (int) Math.pow(2, RECONNECT_MAX_TRIES - connectRetriesRemaining);
+                milliSeconds = Math.min(milliSeconds, backOffDelay);
+                connectRetriesRemaining--;
+            }
+        }
+        checkConnectionTask = scheduler.schedule(() -> checkConnection(), milliSeconds, TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (thing.getStatus() == ThingStatus.ONLINE && (childHandler instanceof DeviceThingHandler)) {
+            logger.debug("childHandlerInitialized() {}", childThing.getUID());
+            try {
+                ResourceReference reference = ((DeviceThingHandler) childHandler).getResourceReference();
+                getClip2Bridge().getResources(reference).getResources().forEach(r -> onResource(r));
+            } catch (ApiException | AssetNotLoadedException e) {
+                // exceptions should not occur here; but log anyway (just in case)
+                logger.warn("childHandlerInitialized() {}", e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Return the application key for the console app.
+     *
+     * @return the application key.
+     */
+    public String consoleGetApplicationKey() {
+        Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+        String applicationKey = config.applicationKey;
+        return "  - Application key: " + (Objects.nonNull(applicationKey) ? applicationKey : "undefined");
```

Comment:
Seems like formatting for a console command? This formatting would fit better in the console class itself.

Reply:
Ok.